udata_croquemort/utils.py

```diff
@@ -0,0 +1,82 @@
+import httplib
+import json
+import logging
+import time
+
+import requests
+
+from flask import current_app
+
+log = logging.getLogger(__name__)
+
+# See http://docs.python-requests.org/en/latest/user/advanced/#timeouts
+# We are waiting 3 sec for the connexion and 9 for the response.
+TIMEOUT = (3.1, 9.1)
+
+DEFAULT_DELAY = 5
+DEFAULT_RETRY = 10
+CONNECTION_ERROR_MSG = 'Unable to reach the URL checker'
+
+ERROR_LOG_MSG = 'Unable to connect to croquemort'
+TIMEOUT_LOG_MSG = 'Timeout connecting to Croquemort'
+
+
+class UnreachableLinkChecker(Exception):
+    pass
+
+
+def is_pending(response):
+    if not response:
+        return True
+    if response.status_code == httplib.NOT_FOUND:
+        return True
+    try:
```

Comment:
Why `try/except` vs. regular `if` check here?

Reply:
Not my code ;-) But yeah, I don't see why not.

Reply:
🎉 Achievement unlocked: I progressed since then 😉

Reply:
Ah ah it was you? 😛

Reply:
Actually `JSONDecodeError` inherits from `ValueError` and this elegant piece of code would catch it when the response is (still) malformed.